src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+					methodContext.push(ret);
+				}
+				break;
+			case INVOKESTATIC:
+				index = ByteUtils.readUnsignedShort(instructions, currentInstruction + 1);
+				ref = (RefInfo) method.owner.runtimeConstantPool[index - 1];
+				targetMethod = ref.getOwner().methods[ref.getID()];
+				if (targetMethod.descriptor.length == 1) {
+					ret = ExecutionEngine.invokeMethodObj(null, targetMethod);
+				} else {
+					Object[] params = new Object[targetMethod.descriptor.length];
+					for (int param = targetMethod.descriptor.length - 1; param > -1; param--) {
+						params[param] = methodContext.pop();
+					}
+					ret = ExecutionEngine.invokeMethodObj(null, targetMethod, params);
+				}
+				if (targetMethod.descriptor[targetMethod.descriptor.length - 1] != PrimitiveType.VOID) {
+					methodContext.push(ret);
+				}
+				break;
+			case INVOKEINTERFACE:
+				break;
+			case INVOKEDYNAMIC:
+				break;
+			case NEW:
+				index = ByteUtils.readUnsignedShort(instructions, currentInstruction + 1);
+				ClassInfo classInfo = (ClassInfo) method.owner.runtimeConstantPool[index - 1];
+				methodContext.push(classInfo.getKlass().newInstance());
+				break;
+			case NEWARRAY:
+				break;
+			case ANEWARRAY:
+				break;
+			case ATHROW:
+				break;
+			case CHECKCAST:
+				break;
+			case INSTANCEOF:
+				break;
+			case MONITORENTER:
+				break;
+			case MONITOREXIT:
+				break;
+			case WIDE:
+				break;
+			case MULTIANEWARRAY:
+				break;
+			case IFNULL:
+				if (methodContext.pop() == null) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFNONNULL:
+				if (methodContext.pop() != null) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case GOTO_W:
+				branchOffset = ByteUtils.readInt(instructions, currentInstruction + 1);
+				break;
+			default:
+				System.out.println("Unsupported instruction");
```

Should probably throw an UnsupportedOperationException here